controllers/component_controller.go

```diff
@@ -649,3 +653,22 @@ func (r *ComponentReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	}).
 		Complete(r)
 }
+
+func (r *ComponentReconciler) incrementCounterAndRequeue(log logr.Logger, ctx context.Context, req ctrl.Request, component *appstudiov1alpha1.Component, componentErr error) (ctrl.Result, error) {
+	if component.GetAnnotations() == nil {
+		component.ObjectMeta.Annotations = make(map[string]string)
+	}
+	count, err := getCounterAnnotation(applicationFailCounterAnnotation, component)
+	if count > 2 || err != nil {
```

Comment:
so we allow up to 3 times retry for component creation due to application not ready failure. and application has 5 retrys?

better put some comment for easy understanding. also, do we want to document those details in book of stonesoup?

Reply:
The difference is because of the delayed requeues. Will add a comment to clarify

Reply:
Updated. 

Don't think these more fine-grained details need to go into the book of stonesoup, we're just slightly delaying when we consider the Component CR to have errored out in this scenario